### PR TITLE
Add a CNI presubmit test job.

### PR DESCRIPTION
**Pull request by bianpengyuan**
Meanwhile unhide postsubmit jobs.

### CODE REVIEW

File: prow/config/jobs/istio.yaml [hunk 1]
```diff
@@ -36,6 +36,18 @@ jobs:
       - name: TEST_SELECT
         value: "-postsubmit,-flaky,-multicluster"
 
+  - name: integ-cni-k8s-tests
+    types: [presubmit]
+    command: [entrypoint, prow/integ-suite-kind.sh, test.integration.pilot.kube.presubmit]
```
Comment (howardjohn) [inline prow/config/jobs/istio.yaml:41]: will we remove `tests/integration/pilot/cni` then?

Comment (bianpengyuan) [reply prow/config/jobs/istio.yaml:41]: Yeah I think we can remove the happy path testing, but I think we'd still need dedicated CNI testing on edge cases such as repair.

**Comment by bianpengyuan** [comment]
@istio/wg-test-and-release-maintainers ptal, thanks!

**Review by brian-avery — APPROVED** [review]
